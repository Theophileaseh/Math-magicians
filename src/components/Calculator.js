import React, { Component } from 'react';
import calculate from '../logic/calculate';
import Button from './Buttons';
import Display from './Display';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: '0',
      next: '',
      operation: '',
    };
  }

  handleClick = (buttonName) => {
    const { total } = calculate(this.state, buttonName);
    const { next } = calculate(this.state, buttonName);
    const { operation } = calculate(this.state, buttonName);
    this.setState({ total, next, operation });
  };

  render() {
    const { total, next, operation } = this.state;

    return (
      <div id="app">
        <Display total={total} next={next} operation={operation} />
        <Button clickHandler={this.handleClick} />
      </div>
    );
  }
}

export default Calculator;
