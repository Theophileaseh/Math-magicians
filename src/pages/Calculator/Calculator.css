.calculatorContainer {
  display: flex;
  padding-top: 80px;
  gap: 60px;
  justify-content: center;
  background: #1a011a;
  height: 100vh;
}

.text {
  font-size: 20px;
  font-weight: 700;
}

.maincalculator {
  border: 2px solid #5f5858;
  padding: 20px;
  background: #121212;
  border-radius: 5px;
  height: max-content;
}

.test-desc {
  color: #fff;
}

@media screen and (max-width: 768px) {
  .calculatorContainer {
    flex-direction: column;
  }

  .text {
    text-align: center;
  }

  .maincalculator {
    width: 80%;
    margin: 0 auto;
  }
}
