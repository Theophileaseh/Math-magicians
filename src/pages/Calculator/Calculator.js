import React from 'react';
import Calculator from '../../components/Calculator';
import './Calculator.css';

const Calculators = () => (
  <div className="calculatorContainer">
    <div className="text">
      <p className="test-desc">Let&apos;s do some math!</p>
    </div>
    <div className="maincalculator">
      <Calculator />
    </div>
  </div>
);

export default Calculators;
