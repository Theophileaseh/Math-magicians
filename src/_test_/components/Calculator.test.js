import React from 'react';
import renderer from 'react-test-renderer';

import Calculator from '../../components/Calculator';

it('matches calculator snapshot', () => {
  const tree = renderer.create(
    <Calculator />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
