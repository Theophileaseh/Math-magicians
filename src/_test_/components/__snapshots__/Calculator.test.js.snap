// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches calculator snapshot 1`] = `
<div
  id="app"
>
  <div
    className="main-screen"
  >
    <span
      className="screen"
    >
      0
    </span>
    <span
      className="screen"
    />
    <span
      className="screen"
    />
  </div>
  <div
    className="buttonContainer"
  >
    <div
      className="row-2"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="button main"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="row-3"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="button main"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="row-4"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="button main"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="row-5"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="button main"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="row-6"
    >
      <button
        className="button double"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="button main"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
